src/lib/alloc.c

```diff
@@ -646,13 +646,13 @@ static void alloc_trace_heap(enum mem_zone zone, uint32_t caps, size_t bytes)
 }
 
 #define DEBUG_TRACE_PTR(ptr, bytes, zone, caps, flags) \
-	do { \
+	if (trace_get()) { \
```

Comment:
I thought about this, but it looks odd to add this check for each caller. Can we implement this inside the tr_xx()? Then we can fix the crash for any other potential callers e.g. the one you are demonstrating in src/init/init.c: 139
```
/* tr_err(&buffer_tr, "CRASHES (and requires a Linux reboot!)"); */
```

Reply:
That would be more complicated because it would have to be done in a few different places because of the number of layers and configurability of the code. 

I don't think it's worth it to guard against a crash that happened only once in years and only in your local workspace where you turned DEBUG_HEAP on _and_ added one extra log statement  in the source. The fewer changes, the better.

`trace_init()` is called extremely early so it's not very demanding not to use the trace before it.

> odd to add this check for each caller.

So far we found two instances, one that I made up and the other one only in your workspace.